front/pages/api/v1/w/[wId]/assistant/conversations/[cId]/messages/index.ts

```diff
@@ -94,10 +95,7 @@ async function handler(
     });
   }
 
-  const conversation = await getConversation(
-    workspaceAuth,
-    req.query.cId as string
-  );
+  const conversation = await getConversation(keyAuth, req.query.cId as string);
```

Comment:
I'm not sure I understand. Getting the conversation should use the workspace auth no?

Also there is no use case for cross workspace retrieval or posting here.

Shouldn't we check that keyAuth workspace matches the URL wId?

Reply:
Thinking more about it we should ~never use the key auth in these endpoints since we want the auth to be scoped to the workspace we're interacting with?

How come we need the keyAuth? Is this due to groups? Do we need to better populate the groups in the workspaceAuth maybe instead?

Reply:
I'm not sure if you've read the PR description and reviewed the code, but there is a strict check at the top of both endpoints to ensure that wId === key.workspaceId. The only difference here is the access to the groups associated with the key. Initially, this was necessary because the implementation of `exchangeSystemKeyForUserAuthByEmail` was incorrect. The groups in the `workspaceAuth` were empty (which was expected). I have since fixed the issue elsewhere. I'll be able to revert this change, but just to clarify, the implementation was secured, and there was no risk of posting to workspaceA using a key from workspaceB.

https://github.com/dust-tt/dust/blob/886902cc2ebb83f25c2404ce2f524376082261c5/front/pages/api/v1/w/%5BwId%5D/assistant/conversations/%5BcId%5D/messages/index.ts#L84-L88

Reply:
Yes yes. Still very scary to switch the auth since it makes the whole authorizarion story very hard to understand for future readers. Hence why I wanted to understand exactly what wasthe plan. In particular you didn't mention in the description that the plan was to revert this.

When is the revert due?

Reply:
(I see it's done)